src/mono/mono/mini/mini.h

```diff
@@ -717,15 +717,15 @@ typedef struct {
 	LLVMArgInfo args [1];
 } LLVMCallInfo;
 
-#define MONO_MAX_SRC_REGS	3
+#define MONO_MAX_SRC_REGS	4
```

Comment:
I believe this is actually not enough and the `3` value is hard-coded in a few different places instead of just using this define.

Reply:
This PR is far away from being complete. I just did the most basic change. Will fix the errors exposed by testing.